### DISCUSSION TITLE
Post-deployment Glider Reports

### DISCUSSION THREAD

**Discussion by prtda** [SWFSC/glider-lab#66] [Ideas]
#### Background
@flaimC and @prtda have been tasked by GW to create a routine post-deployment report for our glider missions. We'll be drawing on examples of previous reporting from @jgwalsh, while incorporating standardized figures, summary data values, quality/performance indices, and other standardized textual information. 

We have a meeting scheduled for Friday 12/6 to discuss with a larger group, but in the meantime (and as we continue to develop an effective reporting format) we can discuss details here, with this conversation hopefully becoming a reference as this develops.

***

##### Email from George quoted here:

> Hi Caleb and Jacob,
> 
> As the subject of this email implies, I have a request for the two of you - borne out of your own success.
> 
> Christian and I have been talking and think it would be very valuable to automate the production of a standard "Deployment Report" for all our glider deployments. I would like to make the eventual pile of such reports available on our website, or at least make them easy for us to provide to folks who are interested in what we are doing - and there are lots of such folks.
> 
> It might be nice for us to schedule a chat next week or the following week to talk through this a bit and maybe develop a sort of outline for such reports. I would be happy for others to join in that conversation. I have some ideas of my own of what should be in such reports, and I'm sure others will have good ideas.
> 
> Maybe you can scour people's calendars and suggest a time when we can discuss. I should add that I am wanting to produce v.0 at the moment and imagine we can increase the version numbers in the future.
> 
> Take care,
> George

**Comment by prtda** [comment]
# Glider Deployment Report | _outline_

## Summary
- Map with track and waypoint plan
- Sensor suite
- Summary statistics
- Timeline
- Mission termination state: 
    - Planned recovery
    - Emergency recovery
    - Unrecovered

## Mission Objectives

## Difficulties
- Failures
- Abnormalities
- Obstacles
- Lessons learned
- etc

## Data
- Overview plots
- Areas of interest
- Preliminary analysis?
- Auxiliary Data
    - SST, SSChl (remote sensing data)
    - Nearby buoys, moorings, shore stations
    - Ship data from cruises (ex. THL)
    - Model nowcasts (ex. WCOFS)

## Glider configurations details

## References for data sources

**Reply by flaimC** [reply]
@prtda  Here's a quick thought on some of the graphics and text that should probably be present. I also wanted to toss out the idea of keeping all reports in a jupyter or quarto book that is rendered as a website on github, examples [here](https://executablebooks.org/en/latest/gallery/). Each project can have a "chapter" and each deployment could have a section under its respective chapter. There could be a intro chapter (or something similar) that links to the most recent reports for the projects.

# Summary
## Map with track and waypoint plan
- Ideally map with bathymetry
- Ice location image if it's an icy year and not in CA
- min/max/avg distance traveled per day
- Denote important locations (deployment/recovery/abort/trouble locations) with different symbols

## Sensor suite
- Diagram of glider with sensor locations
- Sensor calibration dates and data sheets
- Sensor model, output units, sample accuracy and resolution
- Sampling scheme (e.g., diving, hovering climbing) and speed (e.g., 1 hz, 0.5 hz, etc.)

## Summary statistics
- Linearly regress important engineering parameters (e.g., pitch, roll, oil vol, etc.) to look for drift over time
- min/max/mean/std dev/std error of science parameters
- volume of data collected (e.g., X Gb oimagery, Y Gb hydrographic, etc.)
- Number of yos
- Total distance traveled
- Volume of water quantified by data?
- Avg battery usage per dive/day
- Battery % at deployment and recovery
- Average mixed layer depth?

## Timeline
- Date glider deployed
- Dates of important/odd events
- Date glider recovered
- Date data processed/generated

###             •
###             •
###             •

# Data
## Overview plots
- Science time series plots
- Science spatial sections
- Engineering time series with linear regressions
- Colorized TS plots for Science parameters
- Vertical velocity (w) time series --> buoyancy and efficiency estimations?
- Denote profiles, locations of interest

**Reply by prtda** [reply]
This all sounds great! I agree, some sort of notebook-to-website seems like the best idea. I'm partial to Jupyter books, because py. 

Regarding the details, I wonder if we're at the point where we should start digging up papers/reports from other glider users so as not to reinvent too many wheels. Christian probably knows a good place to look for that kind of thing. 

If you want, I can take a crack at a map, I've been trying to learn my way around `cartopy` recently. I have some bathymetry files for Humboldt. I'm working on a map for our other project up here anyways. 

All the full compressed binary files (`.dcd`, `.ecd`?) from calanus are in GCP. I was going to try to use `pyglider` to extract some data from these files but I haven't made it that far yet.

**Reply by flaimC** [reply]
I've had issues in the past with getting catopy to work in notebooks (it fails at building the wheel and needs some special tweaks to get working). I've been using mpl_toolkits [Basemap](https://basemaptutorial.readthedocs.io/en/latest/installation.html) for the rt plots; it plays well with shape files and geopandas. I've heard good things about folium, but have never messed with it.

You can steal this [code](https://github.com/flaimC/slocumRtDataVisTool/blob/de6bff59e4e64758d430636b660ef81deb13576b/rtGliderPlots.py#L931) to read goto files and return a pandas dataframe of coordinates in decimal degrees, if you want.

I wonder how many groups actually do a post deployment report v.s. all of the info just gets put into whatever paper the study is for?

Tomorrow I can make a quick how to for reading in binary files if that would be helpful. The decompression generally seems to work, but I had issues with getting oxygen data when decompressing the binary files.

**Reply by prtda** [reply]
https://underwatergliders.org/glider-best-practices/ 
potentially lots of stuff in here. will I have time to go through it? maybe lol.

thanks for the heads up about cartopy. I'll play with it, and might just not use it in a notebook.

a quick how-to for binary files would be helpful! especially if it's much different from https://pyglider.readthedocs.io/en/latest/getting-started-slocum.html
